controllers/upgrade.go

```diff
@@ -62,6 +62,9 @@ type RollingUpgradeContext struct {
 	metricsMutex        *sync.Mutex
 	DrainTimeout        int
 	IgnoreDrainFailures bool
+	ReplacementNodesMap *sync.Map
+	MaxReplacementNodes int
```

Comment:
So do we allow users to override max replacement nodes per spec?

Reply:
No, we don't. It will remain to be a controller-level flag. Not all the fields in the RollingUpgrade context will end up being a spec in RollingUpgrade object.